app/assets/javascripts/govuk-component/option-select.js

```diff
@@ -43,6 +43,11 @@
       if (this.$optionSelect.data('closed-on-load') == true) {
         this.close();
       }
+
+      var closedStateName = this.$optionSelect.data('closed-state-name');
+      if (closedStateName !== undefined) {
```

Comment:
`hasData` checks for any data set on an element, rather than data for a particular key. In this case, it would return a false positive when there is `closed-on-load` data but no `closed-state-name`.